AppCenterIdentity/AppCenterIdentityTests/MSIdentityTests.m

```diff
@@ -551,6 +551,65 @@ - (void)testSignInFailsWhenNoIdentityConfig {
   [identityMock stopMocking];
 }
 
+- (void)testSecondSignInSuccessAfterFirstSignInFails {
+
+  // If
+  NSString *idToken = @"idToken";
+  NSString *accountId = @"94c82516-cbee-44aa-8a6a-19f8d20322be";
+  id msalResultMock = OCMPartialMock([MSALResult new]);
+  OCMStub([msalResultMock idToken]).andReturn(idToken);
+  OCMStub([msalResultMock uniqueId]).andReturn(accountId);
+  self.sut.clientApplication = self.clientApplicationMock;
+  id identityMock = OCMPartialMock(self.sut);
+  OCMStub([identityMock sharedInstance]).andReturn(identityMock);
+  OCMStub([identityMock canBeUsed]).andReturn(YES);
+
+  // When
+  MSSignInCompletionHandler handler1 = ^(MSUserInformation *_Nullable userInformation, NSError *_Nullable error) {
+    self.signInUserInformation = userInformation;
+    self.signInError = error;
+  };
+  [MSIdentity signInWithCompletionHandler:handler1];
```

Comment:
This one isn't actually using mock instance. You should call `resetInstance` to make `MSIdentity` mock works.

Reply:
Or remove identityMock if you don't need to use it.

Reply:
`resetInstance` is called in the `tearDown` method, that should be sufficient, yes? @jaelim-ms

Reply:
`sharedInstance` is called in `setUp`. That's the problem.

Reply:
Seems it indicates same object even though it shows different addresses.

Reply:
It is because of a partial mock, not a class mock.